services/wallet/collectibles/manager.go

```diff
@@ -315,6 +316,85 @@ func (o *Manager) FetchAllAssetsByOwner(ctx context.Context, chainID walletCommo
 	}
 	return nil, ErrNoProvidersAvailableForChainID
 }
+func (o *Manager) fetchERC1155Balances(ctx context.Context, owner common.Address, chainID walletCommon.ChainID, contractAddress common.Address, tokenIDs []*bigint.BigInt) ([]*bigint.BigInt, error) {
```

Comment:
can u update the router that check balance to leverage this function?

Reply:
done